src/base_alloc/base_alloc_windows.c

```diff
@@ -5,8 +5,38 @@
  * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 */
 
+#include "base_alloc.h"
 #include <windows.h>
 
+static void umf_ba_constructor(void) {}
+
+static void umf_ba_destructor(void) { umf_ba_destroy_global(); }
+
+#ifdef UMF_SHARED_LIBRARY
+
+// constructor and destructor are handled in libumf_windows.c
+
+void umf_ba_init(void) {
+    // do nothing, additional initialization not needed
+}
+
+#else /* UMF_SHARED_LIBRARY */
+
+INIT_ONCE BA_init_once_flag = INIT_ONCE_STATIC_INIT;
+
+BOOL CALLBACK BAinitOnceCb(PINIT_ONCE InitOnce, PVOID Parameter,
+                           PVOID *lpContext) {
+    umf_ba_constructor();
+    atexit(umf_ba_destructor);
```

Comment:
won't we have the same problem with proxy lib as on linux when using atexit?

Reply:
I do not know yet. I have done that exactly like in `libumf_windows.c`.

Reply:
Right, the proxy lib does not support windows yet anyway,